### PR TITLE
Add support for custom kv2 metadata

### LINKED ISSUE #789: Support for custom_metadata in `Versioned.Metadata`
Hi there,
love the Hashicorp Vault support build here! Now I was wondering: can we please have support for the custom_metadata as part of `org.springframework.vault.support.Versioned.Metadata`? I would love to use that in a challenge we are building at OWASP WrongSecrets (See https://github.com/OWASP/wrongsecrets/issues/719 for more details).

Or is there already another way in which I would be able to retrieve the `custom_metadata` for a secrets entry?

### PR DESCRIPTION
**Pull request by commjoen**
This is a first attempt (in Draft, not yet completed) to add custom_metadata to Metadata

Closes #789 

TODO:
 - [x] Agree on the approach written here
 - [x] Update integration tests
 - [x] Add documentation where required

### CODE REVIEW

**Comment by mp911de** [comment]
The changes are pretty straight-forward by adding another property to the already existing types. Once you add tests, we're happy to move forward toward merging this pull request.

File: spring-vault-core/src/test/java/org/springframework/vault/core/VaultVersionedKeyValueTemplateIntegrationTests.java [hunk 1]
```diff
@@ -103,6 +104,34 @@ void shouldCreateVersionedWithCAS() {
...
+		customMetadata.put("foo", "bar");
+		customMetadata.put("uid", "werwer");
+
+		Metadata metadata= Metadata.builder()
+				.customMetadata(customMetadata)
+				.createdAt(Instant.now())
+				.version(Version.unversioned())
+				.build();
+
+		Metadata metadata1 = this.versionedOperations.put(key, Versioned.create(person, metadata));
```
Comment (commjoen) [inline spring-vault-core/src/test/java/org/springframework/vault/core/VaultVersionedKeyValueTemplateIntegrationTests.java:126]: Apologies for asking good sir @mp911de, but this test is failing as the customMetadata seems to be lost. Do you know where I am missing a step?

**Comment by commjoen** [comment]
Hi @mp911de , I think the code is ready for a review.

**Comment by mp911de** [comment]
Looks good, I'll take your pull request from here.

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged and polished now.

**Comment by commjoen** [comment]
Thank you @mp911de !

**Pull request closed without merging**